Demo/Application/Base/BraintreeDemoContainmentViewController.m

```diff
@@ -253,6 +267,18 @@ - (void)containIntegrationViewController:(UIViewController *)viewController {
     return block;
 }
 
+- (void (^)(NSString *paymentMethodNonceString))nonceStringCompletionBlock {
+    static id block;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
```

Comment:
q: Does this prevent retries?

Reply:
May not be important since it's a demo app but having to rebuild the app to retry could make troubleshooting more difficult.

Reply:
Would it be possible to add some comments in this function to give merchants insight into what is happening here when they check out our demo app? I agree that the reason for using `dispatch` is unclear, will probably be unclear to them as well?

Reply:
We can add docs, though I am not sure I have the context to add clear enough documentation to this as dispatch is used rather extensively throughout the SDK currently. Feels a bit weird to add those docs here but nowhere else where we are also using the legacy API in the demo app? It was used by us to match the implementation where we use a nonce as mixing 2 different ways of returning the nonce felt weird.

Reply:
We do use dispatch a lot but `dispatch_once` is the confusing part for me here. I'm sure it made sense at one point in time, but a good question from the uninitiated would be "why is this completion block allowed to execute only once?"

Reply:
My understanding of `dispatch_once` is "perform something once and only once", which is precisely what we're doing and want to do here? We want to get the nonce and ensure the nonce is set as a variable that’s accessible before we execute the next step, which in our case is tapping to make a transaction. We would run into errors if we attempted to retrieve the nonce before this step had completed execution, correct?

This method can be called multiple times (as it is today across all of our ViewControllers). We can run the demo app and make subsistent card transactions (for example) and that calls the completion block and updates the status each time.

Happy to make changes to this, I do think our demo app contains a lot of anti-patterns that I don’t think we’d want merchants to follow. 🙂 Also just trying to wrap my head around this myself!

Reply:
It may be out of scope for this PR but yeah overall that `static` dispatch token isn't bound to any single instance of `BraintreeDemoContainmentViewController`, it's almost equivalent to a `fileprivate var` in Swift.

Reply:
Yeah, I think it's probably out of scope here, but totally agree with asking why on it! I think it brings up a great point about how our demo app modernization with a switch to Swift is probably equally as important as the rest of the SDK since it doesn't represent what we would actually want merchant to do in some cases.